### PR TITLE
New quickstart

### PR DESCRIPTION
**Pull request by gvdongen**
We want to show Durable Execution in the quickstart. This allows us to make the quickstart touch upon the main features of Restate.

### CODE REVIEW

File: templates/java-maven-spring-boot/src/main/java/com/example/restatestarter/Utils.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.example.restatestarter;
+
+// You can remove this file.
+// It's only purpose is providing stubs for the template.
+
+class Utils {
+    public static void sendNotification(String greetingId, String name){
+        if (Math.random() < 0.5) { // 50% chance of failure
+            System.out.println("👻 Failed to send notification: " + greetingId + " - " + name);
+            throw new Error("Failed to send notification: " + greetingId + " - " + name);
```
Comment (slinkydeveloper) [inline templates/java-maven-spring-boot/src/main/java/com/example/restatestarter/Utils.java:10]: throw `RuntimeException`

File: templates/java-maven-spring-boot/src/main/java/com/example/restatestarter/Utils.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        }
+        System.out.println("Notification sent: " + greetingId + " - " + name);
+    }
+
+    public static void sendReminder(String greetingId){
+        if (Math.random() < 0.5) { // 50% chance of failure
+            System.out.println("👻 Failed to send reminder: " + greetingId);
+            throw new Error("Failed to send reminder: " + greetingId);
```
Comment (slinkydeveloper) [inline templates/java-maven-spring-boot/src/main/java/com/example/restatestarter/Utils.java:18]: RuntimeException, fix this in all the other Utils.java files too

File: templates/kotlin-gradle/src/main/kotlin/my/example/Utils.kt
Comment (slinkydeveloper) [inline templates/kotlin-gradle/src/main/kotlin/my/example/Utils.kt:1]: Make the file name lowercase, this is how kotlin people do it with files containing only `fun`

File: templates/java-gradle/src/main/java/my/example/Greeter.java [hunk 3]
```diff
@@ -1,31 +1,28 @@
...
-    return greeting;
+  public String greet(Context ctx, String name) {
+    // Durably execute a set of steps; resilient against failures
+    String greetingId = ctx.random().nextUUID().toString();
+    ctx.run(() -> sendNotification(greetingId, name));
+    ctx.sleep(Duration.ofMillis(1000));
+    ctx.run(() -> sendReminder(greetingId));
+
+    // Respond to caller
+    return "You said hi to " + name + "!";
```
Comment (slinkydeveloper) [inline templates/java-gradle/src/main/java/my/example/Greeter.java:25]: Can you fix the unit test too?

File: templates/java-maven-spring-boot/src/main/java/com/example/restatestarter/Greeter.java [hunk 4]
```diff
@@ -29,8 +23,14 @@ public class Greeter {
...
-  }
+  public String greet(Context ctx, String name) {
+    // Durably execute a set of steps; resilient against failures
+    String greetingId = ctx.random().nextUUID().toString();
+    ctx.run(() -> sendNotification(greetingId, name));
+    ctx.sleep(Duration.ofMillis(1000));
+    ctx.run(() -> sendReminder(greetingId));
 
+    // Respond to caller
+    return "You said " + greetingPrefix + " to " + name + "!";
```
Comment (slinkydeveloper) [inline templates/java-maven-spring-boot/src/main/java/com/example/restatestarter/Greeter.java:34]: Can you fix the unit test too?

File: templates/go/greeter.go [hunk 5]
```diff
@@ -1,14 +1,33 @@
...
 // Greeter is a struct which represents a Restate service; reflection will turn exported methods into service handlers
 type Greeter struct{}
 
-func (Greeter) Greet(ctx restate.Context, greeting string) (string, error) {
-	return fmt.Sprintf("%s!", greeting), nil
+func (Greeter) Greet(ctx restate.Context, name string) (string, error) {
+    //  Durably execute a set of steps; resilient against failures
+	greetingId := restate.Rand(ctx).UUID().String()
+
+	if _, err := restate.Run(ctx, func(ctx restate.RunContext) (restate.Void, error) {
```
Comment (slinkydeveloper) [inline templates/go/greeter.go:15]: I think this should work fine now without `restate.Void`, just

```suggestion
	if _, err := restate.Run(ctx, func(ctx restate.RunContext) error {
```

File: templates/go/greeter.go [hunk 6]
```diff
@@ -1,14 +1,33 @@
...
+		return restate.Void{}, SendNotification(greetingId, name)
+	}); err != nil {
+		return "", err
+	}
+
+	if err := restate.Sleep(ctx, 1*time.Second); err != nil {
+		return "", err
+	}
+
+	if _, err := restate.Run(ctx, func(ctx restate.RunContext) (restate.Void, error) {
```
Comment (slinkydeveloper) [inline templates/go/greeter.go:25]: Same as above

File: templates/python/example.py [hunk 7]
```diff
@@ -1,20 +1,23 @@
...
+from utils import send_notification, send_reminder
+
+greeter = Service("Greeter")
+
 
 @greeter.handler()
 async def greet(ctx: Context, name: str) -> str:
-    return f"Hello {name}!"
+    # Durably execute a set of steps; resilient against failures
+    greeting_id = await ctx.run("generate UUID", lambda: str(uuid.uuid4()))
```
Comment (slinkydeveloper) [inline templates/python/example.py:15]: Opened an issue for this https://github.com/restatedev/sdk-python/issues/30

File: templates/rust-shuttle/Cargo.toml [hunk 8]
```diff
@@ -10,3 +10,4 @@ shuttle-runtime = "0.47.0"
 hyper = { version = "1.4", features = ["http1", "server"] }
 hyper-util = { version = "0.1.7", features = ["http1", "server", "server-graceful"] }
 tracing = "0.1"
+rand = "0.8.5"
```
Comment (slinkydeveloper) [inline templates/rust-shuttle/Cargo.toml:13]: Why you added this? you don't need it

File: templates/rust-shuttle/src/utils.rs [hunk 9]
```diff
@@ -0,0 +1,17 @@
+use rand::random;
+
+pub fn send_notification(greeting_id: &str, name: &str) {
+    if random::<f32>() < 0.5 {
+        println!("👻 Failed to send notification: {} - {}", greeting_id, name);
+        panic!("Failed to send notification: {} - {}", greeting_id, name);
```
Comment (slinkydeveloper) [inline templates/rust-shuttle/src/utils.rs:6]: not a good idea, convert this to error. I'm not even sure we catch panics actually.

Comment (slinkydeveloper) [reply templates/rust-shuttle/src/utils.rs:6]: one thing you can do to make this code better overall is the following:

Change the signature of this function to 

```
pub async fn send_notification(greeting_id: &str, name: &str) -> anyhow::Result<()>
```

And then the context run should be as easy as

```
ctx.run(|| send_notification(&greeting_id, &name)).await?
```

File: templates/rust/src/utils.rs [hunk 10]
```diff
@@ -0,0 +1,17 @@
+use rand::random;
+
+pub fn send_notification(greeting_id: &str, name: &str) {
```
Comment (slinkydeveloper) [inline templates/rust/src/utils.rs:3]: Same comments above

File: templates/python/example.py (hunk 7 above)
Comment (gvdongen) [reply templates/python/example.py:15]: I'll change it once it's there

File: templates/go/greeter.go (hunk 5 above)
Comment (gvdongen) [reply templates/go/greeter.go:15]: that doesn't work for me with v0.12.0

File: templates/python/pydantic_models.py [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+from pydantic import BaseModel
+
+
+# You can also just use a typed dict, without Pydantic
+class GreetingRequest(BaseModel):
+    name: str
+
+
+class Greeting(BaseModel):
+    message: str
```
Comment (slinkydeveloper) [inline templates/python/pydantic_models.py:12]: I would honestly have them in `example.py`, makes the whole thing way more readable, and this is super small...

**Review by slinkydeveloper — COMMENTED** [review]
one minor thing and lgtm

**Pull request merged**